mediatek: filogic: support WAVLINK WL-WN586X3 Rev B

i0ntempest opened the pull request:
This is https://github.com/openwrt/openwrt/pull/19390 rebased on master. Currently work in progress to adopt the common dtsi [here](https://github.com/openwrt/openwrt/blob/main/target/linux/mediatek/dts/mt7981b-wavlink-wl-3port-128nand-common.dtsi).

plappermaul commented:
As said.

- drop the large dts file
- copy the existing one from WL-WN551X over 
- adapt to your needs.

plappermaul reviewed (changes requested):
Thanks. Now only the USB remark left. 

Independant from that: please provide hardware description and installation instructions with this commit. See haukes request on the initial PR https://github.com/openwrt/openwrt/pull/19390#issuecomment-3110369117

plappermaul reviewed (changes requested):
The commit description for a new device must be self contained.

i0ntempest committed:
mediatek: filogic: support WAVLINK WL-WN586X3 Rev B

This commit adds support for the WAVLINK WL-WN586X3 Rev B, which swaps
the 16MB NOR flash for an 128MB NAND flash chip, and uses UBI for
data partition. This commit utilizes the previous submitted dtsi that
includes common hardware for a few Wavlink routers.

Hardware
--------
- SOC: MediaTek MT7981B
- RAM: 256MB DDR3
- Flash: 128MB SPI-NAND (ESMT F50L1G41LB)
- Ethernet: 2x1Gb Lan 1x1Gb Wan
- WiFi: MediaTek MT7981B 2x2 DBDC 802.11ax 2T2R (2.4/5) with 4 external and 1 internal antennae
- LEDs: 2xLan 1x Wan 1x WIFI 1xSTATUS

Installation
------------
Flashing over the vendor WebUI has been tested and unsuccessful (YMMV).
The image need to be flashed via TFTP which can be activated over the onboard UART serial console:
1. Connect UART:  TX->board RX, RX->board TX, GND->board GND.
2. Connect PC to router lan port.
3. Setup the tftp server on PC, set IP to 192.168.10.100.
4. Power on the device. Interrupt boot countdown at uboot boot menu and select 2. Upgrade firmware option.
5. Input the image name and start flashing.

Sample uboot menu:
```
  *** U-Boot Boot Menu ***

      1. Startup system (Default)
      2. Upgrade firmware
      3. Upgrade ATF BL2
      4. Upgrade ATF FIP
      5. Upgrade single image
      6. Load image
      0. U-Boot console
```

NOTE: Do not use the intermediate image here which is for Rev A only:
https://github.com/themaverickdm/firmware-misc/tree/main/wavlink/wl-wn586x3

MAC Addresses
-------------
LAN: 80:3F:5D:xx:xx:x1 (hw, 0x44e, ASCII encoded)
WAN: 80:3F:5D:xx:xx:x2 (hw, 0x460, ASCII encoded)
 2G: 80:3F:5D:xx:xx:x3 (factory, 0x4, raw binary, also on label)
 5G: 80:3F:5D:xx:xx:x3 (Same as 2G)

Signed-off-by: Zhenfu Shi <i0ntempest@i0ntempest.com>
Link: https://github.com/openwrt/openwrt/pull/19785
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

i0ntempest commented:
Looks like release images are not being built for this device, but snapshots do. Anything else needs to be done?
